examples/theming/src/app/layout.tsx

```diff
@@ -0,0 +1,42 @@
+import './global.css'
+import {cookies} from 'next/headers'
+import {BaseStyles, ThemeProvider} from '@primer/react'
+import {StyledComponentsRegistry} from './registry'
+
+export const metadata = {
+  title: 'Next.js',
+  description: 'Generated by Next.js',
+}
+
+export default function RootLayout({children}: {children: React.ReactNode}) {
+  const cookieStore = cookies()
+  const colorPreferenceCookie = cookieStore.get('color-preference')
+  const colorMode = getColorMode(colorPreferenceCookie?.value)
+
+  return (
+    // Note: the focus-visible polyfill adds additional attributes to `html`
+    // that cause hydration mismatch errors
+    <html
+      lang="en"
+      suppressHydrationWarning
+      data-color-mode={colorMode}
+      data-light-theme="light"
+      data-dark-theme="dark"
```

Comment:
Like the idea that this could one day be a helper function

```suggestion
      {...colorThemeAttributes(colorMode)}
```